plugins/registry.js

```diff
@@ -248,6 +281,35 @@ export default class PluginRegistry {
         return id;
     }
 
+    // Register a post sub menu list item by providing some text and an action function.
+    // Accepts the following:
+    // - text - A string or React element to display in the menu
+    // - action - A function to trigger when component is clicked on
+    // - filter - A function whether to apply the plugin into the post' dropdown menu
+    // Returns a unique identifier.
+    registerPostDropdownMenuItem(text, action, filter) {
+        const id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
+            name: 'PostDropdownMenu',
+            data: {
+                id,
+                pluginId: this.id,
+                text: {
+                    id,
+                    text: resolveReactElement(text),
+                    subMenu: [],
+                },
+                action,
+                filter,
+            },
+        });
+
+        const subMenuGroup = new SubMenuGroup(this.id, id, 1);
+        return subMenuGroup.registerMenuItem;
```

Comment:
For consistency, we should also return the id here.

Reply:
I had tried this however since the id is not used external to the structure and actions/filters are mapped to individual menu items,  I do not see the need to also return it.  Is there a use case for the id to be returned?  I apologize for being resistant to this.

Reply:
My gut says we're missing an `unregisterPostDropdownMenuComponent` with which this `id` might be useful to remove the entire post menu. Admittedly, it does seem unlikely to need that, but planning ahead to return multiple parameters would give us more flexibility in the future. Destructuring for this case is fairly painless, IMO.

Reply:
@scottleedavis do you still have open concerns on returning the id from the registry function?

Reply:
I do not @marianunez .  Added.